### DISCUSSION TITLE
How to change the PHP version?

### DISCUSSION THREAD

**Discussion by sovetski** [PHP-CS-Fixer/PHP-CS-Fixer#7118] [Q&A] [answered]
## Support question

Imagine I have a project on 5.3 or 7.3, and I want to use PHP CS FIXER, how can I config it to limit the code changes to PHP 5.3 or 7.3 etc.?

Lets imagine this fake situation: I have 'array()' in my code and php cs fixer transforms it to '[]', but it does not exist on 5.3 or 7.3 (just an example), how to do it? I know that we can use the config to say "dont change it", but is there any global and easy way to do it?

**Comment by keradus** [comment, marked as answer]
process of fixing is followig:
- lint file before fixing , continue if no crash
- modify file knowledge of PHP that PHP runner has (eg list of native methods) , continue if no crash
- lint file after fixing, , continue if no crash
- save file

each of those steps require PHP to fit the expected runtime version. Ie even if you tell that rule to run or not, afterwards PHP linter is based on your current runtime version (it basically runs `php -l ...`), not on expected runtime version (eg cannot run `php -l ...` of php that is available).

for that, we encourage to use php runtime that fits expected php runtime. for your case, to fix codebase that you want to run on PHP 7.3, you should run PHP CS Fixer via PHP 7.3
